### PR TITLE
feat: Add hero section

### PR DESCRIPTION
**Pull request by Sushant203**
# **PR Title**:  
`feat: implement initial homepage hero section`

---

## **Description**:  
This PR introduces the initial implementation of the homepage hero section for the application. The hero section provides an introductory message and branding for Fuse Machines Nepal. It includes a two-column layout with descriptive text and an accompanying image.

---

## **Changes Made**:  
1. Added a new `HeroSection` component:
   - Greeting message with styled headings for emphasis.
   - Placeholder text for additional information.
   - A call-to-action button labeled **"Our Projects"**.
   - An image with a caption to complement the text.
2. Styled the component using Tailwind CSS:
   - Background color: `#0A192F`.
   - Text and spacing utilities for consistent design.
   - Grid layout with responsive design for flexibility.

---

## **To-Do / Pending**:  
- Add responsive styles for mobile and tablet views.
- Include animations for button hover and image fade-in.
- Replace placeholder text with finalized content.
- Optimize `alt` text for accessibility compliance.

---

## **Testing Instructions**:  
1. Pull the branch locally using:
   ```bash
   git pull origin <feature/homepage>

### CODE REVIEW

File: src/components/homepage/HeroSection.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import image from "../../assets/fuse machines.jpeg";
+const HeroSection = () => {
+  return (
+    <section className="bg-[#0A192F] px-6 py-6">
```
Comment (Himalf) [inline src/components/homepage/HeroSection.jsx]: **you can define the colors like primary , secondary colors in tailwind for reuse.**

File: src/components/homepage/HeroSection.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            temporibus quia! Vel sint saepe impedit at quibusdam.
+          </p>
+          {/* buttons */}
+          <button className="border-none bg-[#6F54EF] rounded-md w-fit py-2 px-2 ">
+            Our Projects
+          </button>
+        </div>
+        {/* hero-picture */}
+        <figure>
+          <img src={image} alt="fuse machines nepal" className="rounded-md " />
```
Comment (Himalf) [inline src/components/homepage/HeroSection.jsx:31]: **you can add the meaningful alt text rather than the company name.**

**Review by Himalf — COMMENTED** [review]
### **Feedback for HeroSection Code Review**
#### **Strengths**:
- The component is well-structured and effectively uses Tailwind CSS for styling.
- Semantic tags like `<section>`, `<figure>`, and `<figcaption>` are used correctly.
- The grid layout creates a clean and responsive design.
#### **Improvements**:
- Update the `alt` text for the image to be more descriptive for accessibility.
- Consider making the button dynamic, such as using a `Link` component for navigation.
- Replace the Lorem Ipsum text with meaningful content or placeholders.
#### **Testing**:
- Verify the layout adjusts for smaller screens (grid switching to single-column).
- Add hover effects to the button for better interactivity.

Overall, this is a solid implementation. Great work!

**Comment by nimesh631** [comment]
Great work on the HeroSection component! The layout and design look excellent, and the Tailwind CSS styling is well-implemented. I noticed you've already flagged the placeholder text as a TODO, so this will look even better once finalized content is added. The grid layout is great, but it would be good to add responsiveness for smaller screens using grid-cols-1 or a similar utility class to ensure a seamless experience on all devices. Nice job planning ahead!

**Review by Aayush850 — APPROVED** [review]
### Review Feedback
- Code functionality: Tested and works perfectly.
- Code style: Clean and readable.
- Performance: No issues observed.

Great job on this! Approved.

**Review by Himalf — APPROVED** [review]
**All requested updates have been resolved.**

**Review by nimesh631 — APPROVED** [review]
Looks great and works well! Clean implementation and well-executed design. Approved!

**Pull request merged**